src/Microsoft.ML.Transforms/Text/NgramTransform.cs

```diff
@@ -809,10 +809,14 @@ public sealed class ColumnOptions
             /// <summary>The weighting criteria.</summary>
             public readonly WeightingCriteria Weighting;
             /// <summary>
+            /// Underlying state of <see cref="MaximumTermCounts"/>.
+            /// </summary>
+            private readonly ImmutableArray<int> _maximumTermCounts;
```

Comment:
>_maximumTermCounts [](start = 49, length = 18)

similar question, why not keep the public readonly ImmutableArray, but we're introducing another field?

Reply:
We encourage the uses of `IReadOnlyList` since the new interface of trees was merged.

---
In reply to: [264024489](https://github.com/dotnet/machinelearning/pull/2898#discussion_r264024489) [](ancestors = 264024489)